### ISSUE TITLE
Current project status

### ISSUE THREAD

**Issue by TatriX** [cl21/cl21#91]
What is a current project status? Is it abandoned?

**Comment by jackcarrozzo** [comment]
No more abandoned than usual - I for one use and occasionally commit to this project, and I know I'm not the only one.

**Comment by guicho271828** [comment]
as the project says, this is experimental. think alternatively --- in my opinition this might be a good source of a new library. some ideas could be more purified. but there are things that can only take place in cl21 also.

**Comment by azzamsa** [comment]
Hope this project get any update soon.

**Comment by guicho271828** [comment]
I think many good ideas have been showcased in this repository. To truely make it the next generation common lisp, we need:

+ monthly/quaterly/biquaterly/annual rolling update/versioning
+ feature fixing system / API levelling (stable/unstable)

**Comment by jackcarrozzo** [comment]
It’s kinda ok with me that the repo isn’t changing- I still use it in most of my projects.

**Comment by azzamsa** [comment]
@guicho271828 first, Thank you for all your contribution to community, I have added your application software (non-library) to my awesome-cl-software.

Yes, to make that update/versioning happen, we need full work as what Jonas Bernoulli did to [Magit](https://github.com/magit/magit), He worked in full year with great support from [1987 backers](https://www.kickstarter.com/projects/1681258897/its-magit-the-magical-git-client/updates) .

**Comment by vindarel** [comment]
I also used it with success despite a minor issue (1), but I wouldn't mind this project to show more activity: answer to issues and concerns (2), to simple PRs (3), and in general to hear more about it : ] 

(1) https://github.com/cl21/cl21/issues/99
(2) https://github.com/cl21/cl21/issues/100
(3) https://github.com/cl21/cl21.github.com/pulls simple additions to the website

*(… and to have supporters when I mention it on reddit…)*

**Comment by guicho271828** [comment]
I would add another element that makes this sustainable:

* enough number of contributor/administrator.

A single-maintainer model is not sustainable because if he/she can no longer focus on the project at some point of time, it stagnates. I would like CL21 the result of the community, and for that goal we need a github organization with lots of members in it. Anyone in the organization has a push access and the project may be initially messy but it is ok once the agreement is made, while someone still acts as a benevolent dictator.

**Comment by guicho271828** [comment]
Too often a repo owner forgets to pass the administrator priviledge to one of the collaborators. I experienced this in four projects (macroexpand-dammit, coleslaw, cl-cudd, and this). Many lisp projects tend to have a long lifespan and is usable after some time is passed, but the problem is that the **author** cannot maintain the same responsibility for the same lifespan. The bottleneck is human, rather than the codebase (unlike in the other languages). This has been in my mind for a few month and I am wishing to have time/motivation to write a blog article.

**Comment by azzamsa** [comment]
Absolutely, so that ESR wrote about [In Case of Incapacity](http://www.catb.org/esr/continuity.html).

Mr. Fukamachi still active in github, the last commit [he made](https://github.com/fukamachi/jsonrpc/commit/248ae8c5223ed49de03838b7b84ceefad14d3e94) is 9 days ago, but sadly he had so many projects and repos, with few contributors. 

I really appreciate him, all his project is very useful. of course in the Open Source world, [this is often happened](https://medium.com/@nayafia/how-i-stumbled-upon-the-internet-s-biggest-blind-spot-b9aa23618c58).

At first I wondered why he was doing so many projects, I was just worried about a project being abandoned in the future, but I realized that he put all his efforts and ideas, probably in the hope that one day there would be a lisper that would follow up his project.

Weeks ago, I've contacted him via twitter inbox to ask an advice about Common Lisp, and he's very kind and humble.

**Comment by guicho271828** [comment]
Yes, I talk to him monthly or bimonthly at the lisp meetups.

**Comment by azzamsa** [comment]
what an awesome thing to talk with Lisper around you.

**Comment by vale981** [comment]
I am sorry to bring this back up again, but I do like and appreciate cl21 very much.
Being a novice in CL, I have a rather limited experience and I sure know that some things are just "finished". I wonder if this is the case here (at least it works for me :P).
Also I do wonder if there is a movement in the CL community for modernizing the language in the way that CL21 does and if that is the case, why there isn't more of a push for it.

Maybe this is a case for RTFM but asking does no harm :).

Thanks.

**Comment by azzamsa** [comment]
> Also I do wonder if there is a movement in the CL community for modernizing the language in the way that CL21 does and if that is the case, why there isn't more of a push for it.

what about [SICL](https://github.com/robert-strandh/SICL)

**Comment by jackcarrozzo** [comment]
I use this package from time to time , mostly for the regex features, but I don’t really see a need for the language to be “modernized”. A large reason I use CL in the first place is the consistent syntax and interface; if modernization is to introduce syntactic sugar, then I am against it.

**Comment by vale981** [comment]
Thanks for the tips with SICL :)

**Comment by vindarel** [comment]
> I am sorry to bring this back up again, but I do like and appreciate cl21 very much.

me too so I'll add my 2c.

> Being a novice in CL, I have a rather limited experience 

I was super-excited about CL21 last year as a CL newbie too, now I can do with (more of) CL idiosyncrasies, even more since I found material and since I document stuff (mostly on the [Cookbook !](https://github.com/LispCookbook/cl-cookbook/)). 

> and I sure know that some things are just "finished". I wonder if this is the case here (at least it works for me :P).

it works for me too (I just had two minor issues which I asked on Stack Overflow so one can find them). 
IMO this isn't finished, there are PRs and features requests to address as well as legitimate issues (#100)).

I noted than E. Fukamachi mentioned cl21 in a blog post this year: http://blog.8arrow.org/entry/2018/06/24/214706 but I can't find a translation that makes sense anymore^^

> Also I do wonder if there is a movement in the CL community for modernizing the language in the way that CL21 does and if that is the case, why there isn't more of a push for it.

I don't think so. IMO once one learns some inconsistencies, has decent documentation and is used to importing some packages (alexandria, [lambda shorthands](https://github.com/cbaggers/fn),…), he/she looses interest in this effort.

There are also other packages that are less disturbing than CL21 (rutils).

> Maybe this is a case for RTFM but asking does no harm :)

now this project looks active :]

---

@azzamsa CLIM is a fresh implementation of CL, it'll come with the same peculiar features, I don't think it has the same goal of cl21.

**Comment by vale981** [comment]
> I don't think so. IMO once one learns some inconsistencies, has decent documentation and is used to importing some packages (alexandria, lambda shorthands,…), he/she looses interest in this effort.

After a discussion with a fellow student today, I've come to a similar conclusion. After all the people who standardized CL knew what they were doing. What i find to be particularly great about CL is that you can extend and modify it relatively easily if you aren't happy with it. Especially being able to cherry pick parts of CL21 is cool.

I hope http://phoe.tymoon.eu/clus/doku.php will be going somewhere eventually.

**Comment by vindarel** [comment]
that being said, there's still a lot of room for cl21 IMO, specially, as we said, for CL newcomers just like us not long ago, but ones who will be less patient… and that makes a lot of people !
